crowbar_framework/app/models/service_object.rb

```diff
@@ -1173,6 +1173,10 @@ def apply_role(role, inst, in_queue)
 
       nodes_in_batch = []
 
+      # get databag to remember potential removal of a role
+      databag = Chef::DataBag.load "barclamps/#{@bc_name}"
+      databag["crowbar"]["roles_to_remove"] = []
```

Comment:
we could also add validations, but this would be a fixed attribute then just for that one case (which likely rarely happens)